osu.Game/Screens/Select/Options/BeatmapOptionsOverlay.cs

```diff
@@ -85,45 +83,48 @@ public BeatmapOptionsOverlay()
                     AutoSizeAxes = Axes.X,
                     Origin = Anchor.BottomLeft,
                     Anchor = Anchor.BottomLeft,
-                    Children = new BeatmapOptionsButton[]
-                    {
-                        new BeatmapOptionsRemoveFromUnplayedButton
-                        {
-                            Action = () =>
-                            {
-                                Hide();
-                                OnRemoveFromUnplayed?.Invoke();
-                            },
-                        },
-                        new BeatmapOptionsClearLocalScoresButton
-                        {
-                            Action = () =>
-                            {
-                                Hide();
-                                OnClearLocalScores?.Invoke();
-                            },
-                        },
-                        new BeatmapOptionsEditButton
-                        {
-                            Action = () =>
-                            {
-                                Hide();
-                                OnEdit?.Invoke();
-                            },
-                        },
-                        new BeatmapOptionsDeleteButton
-                        {
-                            Action = () =>
-                            {
-                                Hide();
-                                OnDelete?.Invoke();
-                            },
-                        },
-                    },
                 },
             };
         }
 
+        /// <param name="depth">
+        /// <para>Lower depth to be put on the left, and higher to be put on the right.</para>
+        /// <para>Notice this is different to <see cref="Footer"/>!</para>
+        /// </param>
+        public void AddButton(string firstLine, string secongLine, FontAwesome icon, Color4 colour, Action action, float depth = 0)
+        {
+            buttonsContainer.Add(new BeatmapOptionsButton
+            {
+                FirstLineText = firstLine,
+                SecondLineText = secongLine,
+                Icon = icon,
+                ButtonColour = colour,
+                Depth = depth,
+                Action = () =>
+                {
+                    Hide();
+                    action?.Invoke();
+                },
+            });
+        }
+
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+        {
+            if (args.Repeat) return false;
+            int index = (int)args.Key - (int)Key.Number1;
+            if (index >= 0 && index <= 9)
+            {
+                // Children of ButtonFlow is reversed. The same reason of depth in AddButton.
+                var button = buttonsContainer.Children.Reverse().ElementAtOrDefault(index);
+                if (button != null)
+                {
+                    button.TriggerClick();

```

Comment:
This should probably be handled in the buttons themselves, either by giving them a Key (probably not since we want to have keybinds sometime in the future) or deriving a base class that defines an abstract member for the key (more preferred, imo).